### PR TITLE
Add unit tests for MGO (and slight refactor of controller

### PR DESCRIPTION
**Pull request by csheremeta**
for ease of unit testing) - [OSD-5852](https://issues.redhat.com/browse/OSD-5852)

### CODE REVIEW

**Comment by csheremeta** [comment]
ok  	github.com/openshift/must-gather-operator/pkg/controller/mustgather	0.022s	coverage: 33.7% of statements

File: pkg/controller/mustgather/mustgather_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	recBase := util.NewReconcilerBase(cl, s, cfg, eventRec)
+
+	r := ReconcileMustGather{
+		ReconcilerBase: recBase,
+	}
+
+	req := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name: "test-must-gather",
+			Namespace: "openshift-must-gather-operator",
```
Comment (georgettica) [inline pkg/controller/mustgather/mustgather_controller_test.go:47]: shouldn't this be the name and namespace the mgObj has?
```suggestion
			Name: mgObj.Name,
			Namespace: mgObj.Namespace,
```

File: pkg/controller/mustgather/mustgather_controller.go [hunk 2]
```diff
@@ -194,8 +188,8 @@ func (r *ReconcileMustGather) Reconcile(request reconcile.Request) (reconcile.Re
 	// get operator namespace to manage resources in
 	operatorNs, err := k8sutil.GetOperatorNamespace()
 	if err != nil {
-		log.Error(err, "unable to get operator namespace")
-		return r.ManageError(instance, err)
+		operatorNs = "openshift-must-gather-operator"
+		fmt.Println("using default operator namespace: " + operatorNs)
```
Comment (georgettica) [inline pkg/controller/mustgather/mustgather_controller.go:192]: why shoudn't this be an debug?
```suggestion
		log.V(1).Info("using default operator namespace: " + operatorNs)
```

File: pkg/controller/mustgather/mustgather_controller_test.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+		},
+	}
+
+	res, err := r.Reconcile(req)
+	if err != nil {
+		t.Fatalf("reconcile: (%v)", err)
+	}
+
+	if res != (reconcile.Result{}) {
```
Comment (georgettica) [inline pkg/controller/mustgather/mustgather_controller_test.go:56]: this looks cool, i'll see if this solves our problem in a different place in the code :)

**Review by georgettica — COMMENTED** [review]
small things that might be helpful adding

changed via the UI only and are worth testing locally before merging

aside of that this looks cool! :)

File: pkg/controller/mustgather/mustgather_controller_test.go (hunk 1 above)
Comment (csheremeta) [reply pkg/controller/mustgather/mustgather_controller_test.go:47]: +1

File: pkg/controller/mustgather/mustgather_controller_test.go (hunk 3 above)
Comment (csheremeta) [reply pkg/controller/mustgather/mustgather_controller_test.go:56]: nice! :D hope it helps

File: pkg/controller/mustgather/mustgather_controller.go (hunk 2 above)
Comment (csheremeta) [reply pkg/controller/mustgather/mustgather_controller.go:192]: +1

**Comment by georgettica** [comment]
This looks good and I can +1 generally.
Although I am not working with MGO closely so maybe another set of eyes on it would be helpful

Also worth noting this test structure doesn't use ginkgo/gomega which from my understanding is not what most operators do (not that this code is bad or unclear, just pointing out)

**Comment by georgettica** [comment]
LGTM:
/hold
Until someone who actually develops on MGO can review
Also to make sure it doesn't get merged without an additional reviewer
/approve
As I gave my 2 cents and it looks great now

**Review by fahlmant — APPROVED** [review]
/lgtm

**Pull request merged**